vortex-layout/src/layouts/struct_/eval_stats.rs

```diff
@@ -31,14 +32,15 @@ impl StatsEvaluator for StructReader {
                     .clone()
                     .step_into()
                     .ok_or_else(|| vortex_err!("cannot step into path"))?;
-                futures.push(
+                futures.push_back(
                     self.child(&field)?
                         .evaluate_stats([child_path].into(), stats.clone()),
                 );
             }
         }
 
-        let results = try_join_all(futures)
+        let results = futures
```

Comment:
What's the reason for this part of the diff? Not obvious to me

Reply:
You avoid copying the futures again into futuresordered. That’s roughly how try_join_all works